iox_query/src/exec/gapfill.rs

```diff
@@ -31,17 +37,63 @@ pub struct GapFill {
 }
 
 /// Parameters to the GapFill operation
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub(crate) struct GapFillParams {
     /// The stride argument from the call to DATE_BIN_GAPFILL
     pub stride: Expr,
     /// The source time column
     pub time_column: Expr,
+    /// The origin argument from the call to DATE_BIN_GAPFILL
+    pub origin: Expr,
     /// The time range of the time column inferred from predicates
-    /// in overall the query
+    /// in the overall query
     pub time_range: Range<Bound<Expr>>,
 }
 
+impl GapFillParams {
+    // Extract the expressions so they can be optimized.
+    fn expressions(&self) -> Vec<Expr> {
+        let mut v = Vec::with_capacity(5);
+        v.push(self.stride.clone());
+        v.push(self.time_column.clone());
+        v.push(self.origin.clone());
+        // currently the optimizer rule enforces that both a start and end bound exist,
+        // but we might decide later (for example) to have a default end bound of now().
+        // So be flexible and avoid panicking if either bound does not exist.
+        if let Some(e) = bound_extract(&self.time_range.start) {
+            v.push(e.clone())
+        };
+        if let Some(e) = bound_extract(&self.time_range.end) {
+            v.push(e.clone())
+        };
+        v
```

Comment:
Another way to express this in rust is with the`vec!` macro:

```suggestion
       vec![
        self.stride.clone(),
        self.time_column.clone(), 
        ....
     ]
```